### ISSUE TITLE
ssh_key_name output evaluates to an empty string

### ISSUE THREAD

**Issue by scarolan** [hashicorp-modules/network-aws#16]
I'd hoped to use the same key generated by this module to spin up some other instances besides the HashiStack but alas, the output `ssh_key_name` evaluates to an empty string... :-(

**Comment by scarolan** [comment]
From the outputs:

```
ssh_key_name =
subnet_private_ids = [
    subnet-016304a75861acbc7,
    subnet-0b8eac9782adb24aa,
```

**Comment by scarolan** [comment]
Here's the name that actually appears in the console:

sean-override-8d94a8ea-20180721190802613500000001

**Comment by bensojona** [comment]
@scarolan - planning on getting to this as soon as I can, but may not be until the weekend. Lmk if it's blocking anything.

**Comment by bensojona** [comment]
@scarolan - took a quick look. 

tl;dr

Do not pass in the [`ssh_key_name` and `ssh_key_override` variables](https://github.com/hashicorp/terraform-guides/blob/master/infrastructure-as-code/hashistack/best-practices/terraform-aws/main.tf#L120-L121) to the [`network_aws`](https://github.com/hashicorp/terraform-guides/blob/master/infrastructure-as-code/hashistack/best-practices/terraform-aws/main.tf#L101-L124) module and it should auto generate a key and return the name in the `ssh_key_name` output variable. My suspicion is that you're overriding this with a previously generated key.

More detailed...

Can you confirm you're _not_ setting the [`ssh_key_override`](https://github.com/hashicorp-modules/network-aws/blob/master/variables.tf#L108-L111) variable to `true` and _not_ [passing in](https://github.com/hashicorp-modules/network-aws/blob/master/main.tf#L184) an overridden [ssh_key_name`](https://github.com/hashicorp-modules/network-aws/blob/master/variables.tf#L98-L101)? 

Can you share the Terraform you're using to call it in a gist? Based on the key name that appears in the console it seems as though you're creating an SSH key outside of the module and passing it in (see [HashiStack best-practices provisioning guide](https://github.com/hashicorp/terraform-guides/blob/master/infrastructure-as-code/hashistack/best-practices/terraform-aws/main.tf#L120-L121) for example of where this occurs).

If that is not the behavior you want, you can just remove the `ssh_key_name` and `ssh_key_override` variables from the `network_aws` module and the defaults will auto-generate a key for you and return it in the output variable `ssh_key_name`. That's the intended behavior as you can see in the [HashiStack quick-start provisioning guide](https://github.com/hashicorp/terraform-guides/blob/master/infrastructure-as-code/hashistack/quick-start/terraform-aws/main.tf#L79-L89) where a key is not passed in.

Originally I would output the `ssh_key_name` regardless of whether it was overridden or auto-generated within the module, but I think ultimately removed it as you would generally not need to pull a key name from an output but rather just use the one you're passing in to override. It also seemed more confusing to output a key name as though it had been generated within the module. Curious your thoughts.

Also - if there's a desire for more flexibility in creating or not creating an SSH key, let me know.

Hopefully that adds some clarity, let me know if it's not working as intended.

**Comment by bensojona** [comment]
Closing for now as it should be working as intended.
